querqy-core/src/main/java/querqy/trie/RuleExtractorUtils.java

```diff
@@ -0,0 +1,38 @@
+package querqy.trie;
+
+import querqy.trie.model.ExactMatch;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class RuleExtractorUtils {
```

Comment:
It's hard to understand what this class does and why it is useful. Could you please provide a bit of JavaDoc. Also, the name implies 'rules' - think it's more about dealing with input sequences (cf. my comment on `RuleExtractor`)